### PR TITLE
migrate assume role credentials test

### PR DESCRIPTION
**Pull request by movence**
# Description of changes
Migrate assume role credentials test to GHA. The new terraform `ec2/creds/main` will create a temporary role that `cwa-e2e-iam-role` can assume.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Tested with a forked repo.
```
null_resource.integration_test_run (remote-exec): === RUN   TestAssumeRole
null_resource.integration_test_run (remote-exec): 2023/06/08 21:21:09 Running AssumeRoleTest
...
null_resource.integration_test_run (remote-exec): --- PASS: TestAssumeRole (31.93s)
null_resource.integration_test_run (remote-exec): PASS
null_resource.integration_test_run (remote-exec): ok  	github.com/aws/amazon-cloudwatch-agent-test/test/assume_role	31.932s
```

### CODE REVIEW

File: test/assume_role/agent_configs/config.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{
+  "agent": {
```
Comment (sethAmazon) [inline test/assume_role/agent_configs/config.json:2]: This is not the role_arn for sending metrics to another account?

**Comment by sethAmazon** [comment]
I thought our tests already assume the ec2 role. How does this differ?

**Comment by movence** [comment]
Existing integ tests already use assume role attached as IAM role to an instance (instance profile by using `cwa-e2e-iam-role`). This test is specifically testing STS credentials written on a host, then CWA uses that temporary creds to emit data to CW. The reason this test creates another temporary IAM role is simply to get accessKey/secret/sessionToken by calling `sts assume-role` (temporary IAM role gets `cwa-e2e-iam-role` in its trust policy).

File: test/assume_role/agent_configs/config.json (hunk 1 above)
Comment (movence) [reply test/assume_role/agent_configs/config.json:2]: Are you tagging to a right line? or are you saying there should be another config in agent's config instead having creds file path in `/opt/aws/amazon-cloudwatch-agent/etc/common-config.toml`?

**Review by sethAmazon — APPROVED** [review]

**Comment by sethAmazon** [comment]
Since this tests the common config to set the shared file and profile I do not think we will need a test for perm creds.

File: test/assume_role/agent_configs/config.json (hunk 1 above)
Comment (SaxyPandaBear) [reply test/assume_role/agent_configs/config.json:2]: There is a way to assume role directly for sending metrics or logs to a different account. If this is a lift and shift from the internal test, I don't really care which way we do it.

**Pull request merged**